src/Marker.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  Marker.swift
+//  ios-sdk
+//
+//  Created by Sarah Lensing on 3/22/17.
+//  Copyright © 2017 Mapzen. All rights reserved.
+//
+
+import Foundation
+import TangramMap
+
+public enum MarkerType : Int {
+  case currentLocation, searchPin, routeLine
+}
+
+@objc(MZMarker)
+public class Marker: NSObject {
+
+  private static let kPointStyle = "points"
+  private static let kLineStyle = "lines"
+  private static let kPolygonStyle = "polygons"
+  private static let kDefaultBackgroundColor = UIColor.white
+  private static let kDefaultInteractive = true
+  private static let kDefaultSize = CGSize.zero
```

Comment:
The problem here is a pixel point with a style size defined to be 0 has a specific meaning in Tangram for point labels https://github.com/tangrams/tangram-es/issues/1193. 

Also, specifying a size at all has a meaning, if you have a point sprite (an icon) with no size but it is defined in the stylesheet like so: https://github.com/tangrams/tangram-es/blob/master/scenes/scene.yaml#L45, the size of this default point icon would be used by default (here `32px`x`32px`), unless you specify a `size` parameter in the style. 

So, thinking more of this, It seems to me that the best option is to only provide the size in the style string if it has been provided by the user without assuming a default?